### PR TITLE
Port UnfinishedHeaderBlock to Rust

### PR DESCRIPTION
**Pull request by Rigidity**
From https://github.com/Chia-Network/chia-blockchain/blob/698455ee7ebe8bba16c2acadd0114de5788a9389/chia/types/unfinished_header_block.py

### CODE REVIEW

File: crates/chia-protocol/src/header_block.rs [hunk 1]
```diff
@@ -72,6 +73,18 @@ impl HeaderBlock {
     pub fn first_in_sub_slot(&self) -> bool {
         !self.finished_sub_slots.is_empty()
     }
+
+    pub fn into_unfinished_header_block(self) -> UnfinishedHeaderBlock {
```
Comment (arvidn) [inline crates/chia-protocol/src/header_block.rs:77]: I don't think there's a counterpart to this in `chia-blockchain`, is there?

**Review by arvidn — COMMENTED** [review]
looks good. It's not clear what the new conversion function is intended to be used for however.

File: crates/chia-protocol/src/header_block.rs (hunk 1 above)
Comment (Rigidity) [reply crates/chia-protocol/src/header_block.rs:77]: It's to do this, this helper function isn't needed in the Python code, but it makes sense for the Rust port imo:
https://github.com/Chia-Network/chia-blockchain/blob/main/chia/consensus/block_header_validation.py#L842

Comment (Rigidity) [reply crates/chia-protocol/src/header_block.rs:77]: That's why I didn't write a binding for it

**Review by arvidn — APPROVED** [review]

**Pull request merged**